website/blog/2024-05-24-Agent/index.mdx

```diff
@@ -0,0 +1,142 @@
+- **When talking about multi vs. single agents, it is beneficial to clarify whether we refer to the interface or the architecture.**
+
+I often got asked about two common questions.
+1. What's an agent?
+1. What are pros and cons of multi vs. single agent?
+
+This blog collects my thoughts from several interviews and recent learnings.
+
+## What's an agent?
+
+There are many different types of definitions of agents. When I thought about it when building AutoGen, I was looking for
+the most generic notion that can incorporate all these different types of definitions. And to do that we
+really need to think about the minimal set of concepts that are needed.
+
+In AutoGen, we think about the agent as an entity that can act on behalf of human intent. They can send messages, receive messages, respond to other agents after taking actions and interact with other agents. We think it's a minimal set of capabilities the an agent needs to
+have underneath. They can have different types of backends to support them to perform
+actions and generate replies. Some of the agents can use large models to generate replies. Some other agents can use tools
+underneath to generate tool-based replies and other agents can use human input as a way to reply to
+other agents. And you can also have agents that mix these different types of backends or have more
+complex agents that have internal conversations among multiple agents. But on the surface, other
+agents still perceive it as a single entity to communicate to.
+
+With this definition, we can incorporate both very simple agents that can solve simple tasks with
+a single backend, but also we can have agents that are composed of multiple simpler agents. One can recursively build up more powerful agents. The
+agent concept in AutoGen can cover all these different complexities.
+
+## What are pros and cons of multi vs. single agent?
+
+This question is asked in a variety of ways.
+> Why shall I use multiple agents instead of a single agent?
+
+> Why think about multi agents when we don't have a strong single agent?
+
+> Does multi-agent increase the complexity, latency and cost?
+
+> When shall I use multi-agents vs. single agent?
+
+When we use the word multi-agent and single-agent, I think there
+are at least two different dimensions we need to think about.
+
+* Interface. That means from the user's point of view, do they interact with the system in a single
+interaction point or do they see explicitly multiple agents working and need to interact with multiple of
+them?
+* Architecture. Are there multiple agents underneath running at the backend?
+
+A particular system can have a single-agent interface and a multi-agent architecture, but the users don't need to know that.
+
+### Interface
+
+In some application it's better to have a single interaction point for
+the user to simplify it. But in other applications, for example, when the application is about having multiple
```

Comment:
When you say "simplify it", are you referring to simplifying the application or the interaction?